Dockerfile

```diff
@@ -1,15 +1,19 @@
 FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12 AS builder
+ARG TAGS=""
 WORKDIR /go/src/github.com/openshift/machine-config-operator
 COPY . .
 # FIXME once we can depend on a new enough host that supports globs for COPY,
 # just use that.  For now we work around this by copying a tarball.
 RUN make install DESTDIR=./instroot && tar -C instroot -cf instroot.tar .
 
 FROM registry.ci.openshift.org/ocp/4.12:base
+ARG TAGS=""
 COPY --from=builder /go/src/github.com/openshift/machine-config-operator/instroot.tar /tmp/instroot.tar
 RUN cd / && tar xf /tmp/instroot.tar && rm -f /tmp/instroot.tar
 COPY install /manifests
-RUN if ! rpm -q util-linux; then yum install -y util-linux && yum clean all && rm -rf /var/cache/yum/*; fi
+RUN if [[ "${TAGS}" == "fcos" ]]; then sed -i 's/rhel-coreos-8/fedora-coreos/g' /manifests/*; \
+    elif [[ "${TAGS}" == "scos" ]]; then sed -i 's/rhel-coreos-8/centos-stream-coreos-9/g' /manifests/*; fi && \
```

Comment:
This looks totally fine as is but I think it'd be a bit nicer to have this be part of the `Makefile` flow so it's easier to do in local builds.  IOW as is right now a local `make` will always do RHCOS, but it'd be nicer to support e.g. `env OS=scos make` to get e.g. a `machine-config-daemon` binary I can just copy to a node and run live without doing the whole container flow.

(But again, fine by me as is!)

Reply:
Good point! I'll take another look :)